packages/core/src/asciidoctor-core-api.js

```diff
@@ -3047,3 +3047,402 @@ Opal.Asciidoctor.SyntaxHighlighterBase = SyntaxHighlighterBase
+}
+
+/**
+ * Check if the table has a header
+ * @returns {boolean}
+ */
+Table.prototype.hasHeader = function () {
+  return (this.rows !== Opal.nil) && (this.rows.head.length > 0)
+}
+
+/**
+ * Check if the table has a header option set
+ * @returns {boolean}
+ */
+Table.prototype.hasHeaderOption = function () {
+  return this.has_header_option
+}
+
+/**
+ * Check if the table has a footer
+ * @returns {boolean}
+ */
+Table.prototype.hasFooter = function () {
+  return (this.rows !== Opal.nil) && (this.rows.foot.length > 0)
+}
+
+/**
+ * Check if the table has the footer option set
+ * @returns {boolean}
+ */
+Table.prototype.hasFooterOption = function () {
+  return this.hasOption('footer')
+}
+
+/**
+ * Check if the table has the autowidth option set
+ * @returns {boolean}
+ */
+Table.prototype.hasAutowidthOption = function () {
+  return this.hasOption('autowidth')
+}
+
+/**
+ * Get the number of rows in the table. Please be carefull, the
+ * header and footer rows are also counted
+ * @returns {string|undefined}
+ */
+Table.prototype.getRowcount = function () {
+  return this.getAttribute('rowcount')
+}
+
+/**
+ * Get the number of columns in the table
+ * @returns {string|undefined}
+ */
+Table.prototype.getColcount = function () {
+  return this.getAttribute('colcount')
+}
+
+// Table Column
+
+/**
+ * Methods to manage the columns of an AsciiDoc table. In particular, it
+ * keeps track of the column specs
+ * @namespace
+ * @extends AbstractNode
+ */
+var Column = Opal.Asciidoctor.Table.Column
+
+/**
+ * Create a new Column element
+ * @param {Table} table
+ * @param {number} index
+ * @param {Object|undefined} attributes
+ * @returns Column
+ */
+Column.create = function (table, index, attributes) {
+  let ret = this.$new(table, index, attributes)
+  return ret
+}
+
+/**
+ * Get the attribute with the given name. If the attribute is not
+ * set, undefined will be returned
+ * @param {string} name the name of the attribute
+ * @returns {string|undefined}
+ */
+Column.prototype.getAttribute = function (name) {
+  let ret = this.getAttributes()[name]
+  return (ret !== Opal.nil) ? ret : undefined
+}
+
+/**
+ * Get the colnumber of this cell
+ * @returns {string|undefined}
+ */
+Column.prototype.getColnumber = function () {
+  return this.getAttribute('colnumber')
+}
+
+/**
+ * Get the width of this cell
+ * @returns {string|undefined}
+ */
+Column.prototype.getWidth = function () {
+  return this.getAttribute('width')
+}
+
+/**
+ * Get the hAlign of this cell
+ * @returns {string|undefined}
+ */
+Column.prototype.getHAlign = function () {
+  return this.getAttribute('halign')
+}
+
+/**
+ * Get the vAlign of this cell
+ * @returns {string|undefined}
+ */
+Column.prototype.getVAlign = function () {
+  return this.getAttribute('valign')
+}
+
+/**
+ * Get the style of this cell
+ * @returns {string}
+ */
+Column.prototype.getStyle = function () {
+  let ret = this.style
+  return (ret !== Opal.nil) ? ret : undefined
```

Comment:
Could you please use the following format (for consistency):

```suggestion
  var style = this.style
  return style === Opal.nil ? undefined : style
```

So:
- `var` instead of `let` (we might use `let` and `const` soon but for now we stick with `var` mostly to maintain compatibility)
- use the name of the method as the name of the variable (`this.style` -> `var style`)
- compare the variable to `Opal.nil` and return undefined or the value (`style === Opal.nil ? undefined : style`) without parenthesis.

Reply:
That may be in more than one place. I'll check it before committing